### PR TITLE
UPSTREAM: <carry>: openshift: Set machine addresses

### PR DESCRIPTION
**Pull request by ingvagabund**
Set hostname, private IP, public IP and external DNS name in machine.Status.Addresses.
The list of addresses is needed to approve a node and link a node with
it's corresponding machine.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 1]
```diff
@@ -150,6 +152,83 @@ func (s *Reconciler) Update(ctx context.Context) error {
 		}
 	*/
 
+	networkAddresses := []apicorev1.NodeAddress{}
+
+	// The computer name for a VM instance is the hostname of the VM
+	// TODO(jchaloup): find a way how to propaget the hostname change in case
```
Comment (frobware) [inline pkg/cloud/azure/actuators/machine/reconciler.go]: typo: propagate.

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 2]
```diff
@@ -150,6 +152,83 @@ func (s *Reconciler) Update(ctx context.Context) error {
...
+				VnetName: azure.GenerateVnetName(s.scope.Cluster.Name),
+			})
+			if err != nil {
+				klog.Errorf("Unable to get %q network interface: %v", ifaceName, err)
+				continue
+			}
+
+			niface, ok := networkIface.(network.Interface)
+			if !ok {
+				klog.Errorf("Unable to cast %q to network.Interface: %v", ifaceName, err)
```
Comment (frobware) [inline pkg/cloud/azure/actuators/machine/reconciler.go]: Report what the actual type turned out to be (via `%T`). Would help debugging if we run into this.

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 3]
```diff
@@ -150,6 +152,83 @@ func (s *Reconciler) Update(ctx context.Context) error {
...
+					if ipConfig.PrivateIPAddress != nil {
+						if ipConfig.PrivateIPAddress != nil {
+							networkAddresses = append(networkAddresses, apicorev1.NodeAddress{
+								Type:    apicorev1.NodeInternalIP,
+								Address: *ipConfig.PrivateIPAddress,
+							})
+						}
+					}
+
+					if ipConfig.PublicIPAddress != nil {
```
Comment (frobware) [inline pkg/cloud/azure/actuators/machine/reconciler.go]: nit: check for `nil` and `continue`. Would help with the indentation.

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 4]
```diff
@@ -150,6 +152,83 @@ func (s *Reconciler) Update(ctx context.Context) error {
...
+						if ipConfig.PublicIPAddress.ID != nil {
+							publicIPInterface, publicIPErr := s.publicIPSvc.Get(ctx, &publicips.Spec{Name: path.Base(*ipConfig.PublicIPAddress.ID)})
+							if publicIPErr != nil {
+								klog.Errorf("Unable to get %q public IP: %v", path.Base(*ipConfig.PublicIPAddress.ID), publicIPErr)
+								continue
+							}
+
+							ip, ok := publicIPInterface.(network.PublicIPAddress)
+							if !ok {
+								klog.Errorf("public ip get returned invalid network interface")
```
Comment (frobware) [inline pkg/cloud/azure/actuators/machine/reconciler.go]: Report what the actual type turned out to be (via %T). Would help debugging if we run into this.

File: pkg/cloud/azure/actuators/machine/reconciler.go (hunk 2 above)
Comment (frobware) [reply pkg/cloud/azure/actuators/machine/reconciler.go]: Also logging `err` here seems wrong.

Comment (frobware) [reply pkg/cloud/azure/actuators/machine/reconciler.go]: And the message contains `ifaceName` which also seems wrong; suspecting copy/paste here :)

Comment (ingvagabund) [reply pkg/cloud/azure/actuators/machine/reconciler.go]: `ifaceName` is intended, it's name of the interface

**Comment by enxebre** [comment]
/lgtm
In follow ups we'll want to:
Set internalDNS to satisfy machine approver
Go async and track instance status https://github.com/openshift/cluster-api-provider-gcp/pull/23

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 5]
```diff
@@ -153,6 +155,82 @@ func (s *Reconciler) Update(ctx context.Context) error {
...
+				VnetName: azure.GenerateVnetName(s.scope.Cluster.Name),
+			})
+			if err != nil {
+				klog.Errorf("Unable to get %q network interface: %v", ifaceName, err)
+				continue
+			}
+
+			niface, ok := networkIface.(network.Interface)
+			if !ok {
+				klog.Errorf("Network interfaces get returned invalid network interface, getting %T instead", networkIface)
```
Comment (frobware) [inline pkg/cloud/azure/actuators/machine/reconciler.go]: This should log (for %T) `niface`.

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 6]
```diff
@@ -153,6 +155,82 @@ func (s *Reconciler) Update(ctx context.Context) error {
...
+
+			for _, ipConfig := range *niface.InterfacePropertiesFormat.IPConfigurations {
+				if ipConfig.PrivateIPAddress != nil {
+					networkAddresses = append(networkAddresses, apicorev1.NodeAddress{
+						Type:    apicorev1.NodeInternalIP,
+						Address: *ipConfig.PrivateIPAddress,
+					})
+				}
+
+				if ipConfig.PublicIPAddress != nil && ipConfig.PublicIPAddress.ID != nil {
```
Comment (frobware) [inline pkg/cloud/azure/actuators/machine/reconciler.go]: Can this not reduce to: 

```go
if ipConfig.PublicIPAddress == nil || ipConfig.PublicIPAddress.ID == nil {
        continue
}
```

which I think would help the indentation.

**Pull request merged**